serverless/src/exportSearch/handler.js

```diff
@@ -1,64 +1,106 @@
+import crypto from 'node:crypto'
+
+import AWS from 'aws-sdk'
 import axios from 'axios'
+import { get } from 'sendero'
 
-import { determineEarthdataEnvironment } from '../util/determineEarthdataEnvironment'
-import { getApplicationConfig, getEarthdataConfig } from '../../../sharedUtils/config'
-import { getJwtToken } from '../util/getJwtToken'
-import { parseError } from '../../../sharedUtils/parseError'
+import { getEarthdataConfig } from '../../../sharedUtils/config'
+import doesBucketExist from '../util/aws/doesBucketExist'
+import { getS3Config } from '../util/aws/getS3Config'
+import { getSearchExportBucket } from '../util/aws/getSearchExportBucket'
+import { getDbConnection } from '../util/database/getDbConnection'
 import { getEchoToken } from '../util/urs/getEchoToken'
-import { prepareExposeHeaders } from '../util/cmr/prepareExposeHeaders'
-import { jsonToCsv } from './jsonToCsv'
+import retry from '../util/retry'
+import { jsonToCsv } from '../util/jsonToCsv'
 import { wrapAxios } from '../util/wrapAxios'
 
 const wrappedAxios = wrapAxios(axios)
 
+// AWS S3 adapter
+let s3
+
 /**
- * Perform a loop through collection results and return the full results in the requested format.
+ * Perform a loop through collection results and upload the full results in the requested format to S3
  * @param {Object} event Details about the HTTP request that it received
  */
 const exportSearch = async (event, context) => {
-  // https://stackoverflow.com/questions/49347210/why-aws-lambda-keeps-timing-out-when-using-knex-js
-  // eslint-disable-next-line no-param-reassign
-  context.callbackWaitsForEmptyEventLoop = false
+  try {
+    // https://stackoverflow.com/questions/49347210/why-aws-lambda-keeps-timing-out-when-using-knex-js
+    // eslint-disable-next-line no-param-reassign
+    context.callbackWaitsForEmptyEventLoop = false
+
+    if (process.env.IS_OFFLINE) {
+      // fake/mock values when doing local development
+      AWS.config.update({
+        accessKeyId: 'MOCK_ACCESS_KEY_ID',
+        secretAccessKey: 'MOCK_SECRET_ACCESS_KEY',
+        region: 'us-east-1'
+      })
+    }
 
-  const { body, headers } = event
+    const dbConnection = await getDbConnection()
 
-  const { defaultResponseHeaders } = getApplicationConfig()
+    // we set the concurrency to 1 in aws-functions.yml, so we only process one at a time
+    // this way, if there's an issue, we can retry only the one that failed
+    const sqsRecord = event.Records[0]
 
-  const { data, requestId } = JSON.parse(body)
+    if (!sqsRecord) return
 
-  const { format, variables, query } = data
+    const { body } = sqsRecord
 
-  const earthdataEnvironment = determineEarthdataEnvironment(headers)
+    const { extra, params } = JSON.parse(body)
 
-  const requestHeaders = {
-    'X-Request-Id': requestId
-  }
+    const {
+      earthdataEnvironment, filename, key, jwt, requestId, userId
+    } = extra
+    if (!filename) throw new Error('missing filename')
```

Comment:
I think a new line would be nice in between your `const` declaration and the `if` statements

Reply:
done with https://github.com/nasa/earthdata-search/pull/1630/commits/3af3f6dac6354677f8442bfd1f96be477d16b72a